### ISSUE TITLE
Problems with colour profiles when using LEDs, out-of-gamut scenarios

### ISSUE THREAD

**Issue by goblin** [Beep6581/RawTherapee#3626]
Hi,

I'm on a self-compiled RT 5.0.

When shooting pics of my LED-lit juggling balls, and overexposing them a lot, or pushing the RAWs to extend the tonal range, I'm experiencing very weird things going on with colours. Abrupt, sharp changes happen and random, non-existent colours pop-up.

An example can be seen with the ARW at http://yatima.uukgoblin.net/rt_debug/DSC01111.ARW. The colour profile from Adobe DNG is supplied at http://yatima.uukgoblin.net/rt_debug/Sony_SLT-A77_Adobe_Standard.dcp. Some screenshots at http://imgur.com/a/KLHTe. This is a picture of my lit blue juggling ball on a reddish towel background.

You can see in the screenshots (or in your copy of RT) that when either the "Camera standard" profile or the special Sony_SLT-A77_Adobe_Standard.dcp profile is selected, the colours look very weird - they sharply transition from one hue of blue to another, or to purple, which didn't exist there at all. This is more prominent when trying to push the exposure up, either with the Exposure compensation slider, or the Tone curve 1 (the curve can be seen in the screenshot titled "2. Upped gamma, custom DCP").

The scene looks very natural only when "no profile" is selected, although in that case the colour is a bit too greenish compared to what was seen in real life (the LED was blue-only). But there are no abrupt changes or nothing weird going on even when upping the exposure.

Why do you think this happens? Why do the colour profiles, which are meant to improve the colours of a scene, work so badly in this lighting scenario? What can be done to achieve a balance between good colours while having no funny purples showing up with sharp edges?

Please feel free to close the issue if you think it's not related to RT (but, e.g., to the colour profiles). I'm raising mostly as a request for help, as I'm not sure if I'm doing something wrong or whether I should hack the colour profiles (how?) or something else.

Thanks!

**Comment by Benitoite** [comment]
Could this be because the profile was generated for reflectance and not emissive spectrum?

**Comment by goblin** [comment]
> Could this be because the profile was generated for reflectance and not emissive spectrum?

Yup, that's a very likely cause. However I've no idea where to find a profile that'd work in my case. Even the in-camera processing has problems with those LEDs.

All the articles I could find about creating your own profiles mention shooting a calibrated colour chart under a D65 illuminant and stuff like that. This clearly won't work in cases such as this. There would need to be a way to ensure that after such calibration, the profile doesn't do something crazy with colours it didn't encounter before (like the ones coming from the LEDs).

There's gotta be someone who knows more about this though, I'm very new to all this ;-)

**Comment by goblin** [comment]
http://yatima.uukgoblin.net/rt_debug/DSC01106.ARW is another, earlier example. Beep6581 suggested using the profile from Adobe DNG and came up with this result: https://i.imgur.com/dHmDbAf.jpg. It did help to fix some blues from the "Camera standard" profile as seen here: https://i.imgur.com/XEXgpJQ.jpg. But those blues are still a little off, they have a rather sharp jump from the well-lit area (where it's kinda greeny) and the less-well-lit one, where it becomes more blue.

This also shows a problem with the red LED, on the top. It becomes yellow when overexposed. There should be no yellow light there at all. It doesn't look as bad as the other problems, as perhaps perceptually over-exposed red can become yellow, but it's not right either, IMO.

**Comment by Benitoite** [comment]
Interesting results. Did you try changing the highlight recovery method? I have experienced weird LED highlights without touching color profiles.

**Comment by goblin** [comment]
> Interesting results. Did you try changing the highlight recovery method? I have experienced weird LED highlights without touching color profiles.

Nope, not sure where that is? For the reference image, it must have been set as per default, as I started with the Neutral processing profile.

**Comment by Benitoite** [comment]
It towards the top of the exposure section.

**Comment by Benitoite** [comment]
I had a dream last night about shooting the color chart on a calibrated LED type monitor to get the emissive spectrum. Wonder if those types of profiles are out there?

**Comment by goblin** [comment]
The highlight recovery does produce slightly better results sometimes, especially with the "luminance recovery" option. Thanks, I'll keep trying it!

The result is still not quite perfect.

Interesting dream! I wonder about that too, now ;-)

**Comment by Benitoite** [comment]
I've been doing some reading up... perhaps shoot a transparency color target like IT8.7/1 on an LED light table and generate an emissive profile from that would allow for the proper gamut compression needed?

**Comment by goblin** [comment]
OK, it's been a while, but I still have those types of problems ;-)

On RawTherapee 5.3, the default "Camera standard" color now looks even worse. The "No proflie" setting on Input Profile is still the best in terms of smoothness (but of course not so good when it comes to color reproduction)

Unfortunately I don't know exactly what you mean in your last comment, Benitoite. Are you saying I should take some calibrated color sheet, and then photograph it with some of the LED lights? Where would I even begin creating a profile, is there any software for that? Wouldn't using different LEDs break it all?
